streams/src/commonMain/kotlin/com/mirego/trikot/streams/reactive/processors/SwitchMapProcessor.kt

```diff
@@ -65,6 +67,12 @@ class SwitchMapProcessor<T, R>(parentPublisher: Publisher<T>, private var block:
             onNextValidation.setOrThrow(1, 0)
         }
 
+        override fun onError(t: Throwable) {
+            super.onError(t)
+            cancellableManagerProvider.cancelPreviousAndCreate()
```

Comment:
wondering why we need to recreate the cancellableManager after an `onError`

Reply:
`CancellableManagerProvider` only have `cancelAndCreate` it does not have `cancel` only.

Reply:
That's on purpose I guess ?

Reply:
Indeed